src/sdk/button.iframe.ts

```diff
@@ -0,0 +1,81 @@
+/**
+ * LikeButtonSdk Constructor
+ * @config liker info likerId:string ref:HTMLElement | string  href?:string
+ */
+interface LikeButtonConfig {
+  likerId: string;
+  ref: HTMLElement;
+  href?: string;
+  likeStyleElement?: HTMLElement;
+}
+
+class LikeCoinButton {
+  likerId: string;
+
+  ref: HTMLElement;
+
+  href: string;
+
+  likeStyleElement!: HTMLElement;
+
+  constructor(config: LikeButtonConfig) {
+    if (!config || !config.likerId) throw new Error('Missing config.likerId');
+    this.likerId = config.likerId;
+    this.ref = config.ref;
+    this.href = config.href || (window && window.location.href) || '';
+    this.insertStyle();
+  }
+
+  mount() {
+    if (this.ref instanceof HTMLElement === false) {
+      this.ref = document.querySelector(`${this.ref}`) as HTMLElement;
+    }
+    if (document.querySelector('.likecoin-embed')) return;
+    // set like user info
+    this.ref.classList.add('likecoin-embed', 'likecoin-button');
+    this.ref.setAttribute('data-liker-id', this.likerId);
+    this.ref.setAttribute('data-href', this.href);
+
+    this.href = encodeURIComponent(this.href);
+    const src = `https://button.like.co/in/embed/${this.likerId}/button?referrer=${this.href}`;
+    this.ref.textContent = '';
+    this.ref.appendChild(document.createElement('div'));
+    const iframe = document.createElement('iframe');
+    // iframe.onload = () => {
+    //   console.log('load');
+    // };
+    // iframe.onerror = err => {
+    //   console.log('err', err);
+    // };
+    iframe.setAttribute('src', src);
+    iframe.setAttribute('frameborder', '0');
+    iframe.setAttribute('scrolling', 'no');
+    this.ref.appendChild(iframe);
+  }
+
+  insertStyle() {
+    this.likeStyleElement = document.createElement('style');
+    this.likeStyleElement.innerHTML = `
+              .likecoin-button {
+              margin-right: 0;
+              position: relative;
+              width: 100%;
+              max-height: 187px;
```

Comment:
Not understand why max-height should be constrained to 187px
If you want to limit the size of the button, should use `max-width` instead of `max-height`